content/300-guides/500-other/880-troubleshooting-orm/100-help-articles/700-netlify-caching-issue.mdx

```diff
@@ -0,0 +1,97 @@
+---
+
+## Problem
+
+If you deploy an application using Prisma to [Netlify](https://netlify.com/), you may run into the following error message on deployment:
+
+```
+Prisma has detected that this project was built on Netlify, which caches dependencies. This leads to an outdated Prisma Client because Prisma's auto-generation isn't triggered. To fix this, make sure to run the \`prisma generate\` command during the build process.
+
+Learn how: https://pris.ly/d/netlify-build
+```
+
+This occurs because Netlify caches the dependencies of your project until one of those dependencies changes. It does this to allow faster builds, and while this is typically a good thing, it causes some problems for Prisma Client.
+
+Prisma uses a `postinstall` hook to generate Prisma Client when dependencies are installed. Because Netlify uses cached modules, this `postinstall` hook never gets run in subsequent deployments after the initial deployment. This results in Prisma Client becoming out of sync with your database schema.
+
+This error message prevents this situation from happening and directs you here to learn how to fix the root issue.
+
+<details><summary>Prisma Client versions below 4.13.0</summary>
+
+On Prisma Client versions lower than 4.13.0, you may encounter error messages that look like the following:
+
+```
+// 1: When adding a field:
+Unknown arg `name` in data.name for type UserCreateInput. Did you mean `nick`?
+
+// 2: When removing a field:
+Invalid `prisma.user.create()` invocation:\n\n\nThe column `User.name` does not exist in the current database.
+
+// 3: When a model was removed/renamed
+Invalid `prisma.user.deleteMany()` invocation:\n\n\nThe table `public.User` does not exist in the current database.
+
+// 4: When a model was added
+Cannot read properties of undefined (reading 'create')
+```
+
+The solutions described in this guide are meant to solve these problems.
+
+</details>
+
+## Solution
+
+This issue can be solved by explicitly generating Prisma Client on every deployment. Running `npx prisma generate` before each deployment will ensure Prisma Client is up-to-date.
+
+You can configure the deployment to run this command in multiple different ways:
+
+### A custom `postinstall` script
+
+<Admonition type="info">
+
+This is the preferred method as it is a universal solution.
+
+</Admonition>
+
+Within the `scripts` section of your project's `package.json` file, if there is not already a script named `postinstall`, add one and add `npx prisma generate` in that script:
+
+```json highlight=4;add
+{
+  ...
+  "scripts" {
+    "postinstall": "npx prisma generate"
+  }
+  ...
+}
+```
+
+### The application's `build` script in `package.json`
+
+Within the `scripts` section of your project's `package.json` file, within the `build` script, prepend `npx prisma generate` to the existing build command:
+
+```json highlight=4;add
+{
+  ...
+  "scripts" {
+    "build": "npx prisma generate && <actual-build-command>"
+  }
+  ...
+}
+```
```

Comment:
I think this looks better than the Vercel suggestion.

(Still could use the concrete Next.js example as that will often be what is actually needed)

Reply:
TODO: look into this after the release